api/test/unit/utils/test_photon.py

```diff
@@ -216,76 +218,136 @@ def raise_exc(*args, **kwargs):
+        (requests.ReadTimeout("whoops"), "requests.exceptions.ReadTimeout"),
+        (requests.Timeout("whoops"), "requests.exceptions.Timeout"),
+        (requests.exceptions.SSLError("whoops"), "requests.exceptions.SSLError"),
+        (OSError("whoops"), "builtins.OSError"),
+    ],
+)
+@alert_count_params
+def test_get_exception_handles_error(
+    exc,
+    exc_name,
+    count_start,
+    should_alert,
+    capture_exception,
+    setup_requests_get_exception,
+    redis,
 ):
-    exc = requests.ReadTimeout()
     setup_requests_get_exception(exc)
-    key = f"{settings.THUMBNAIL_TIMEOUT_PREFIX}subdomain.example.com"
-    redis.set(key, 5)
+    month = get_monthly_timestamp()
+    key = f"thumbnail_error:{exc_name}:subdomain.example.com:{month}"
+    redis.set(key, count_start)
 
-    with pytest.raises(UpstreamThumbnailException, match=r"due to timeout"):
+    with pytest.raises(UpstreamThumbnailException):
         photon_get(TEST_IMAGE_URL)
 
-    capture_exception.assert_called_once_with(exc)
-
-    assert redis.get(key) == b"6"
-
-
-def test_get_http_error_pass_threshold(
-    capture_exception, setup_requests_get_exception, redis
-):
-    exc = requests.HTTPError()
-    setup_requests_get_exception(exc)
-    key = f"{settings.THUMBNAIL_HTTP_ERROR_PREFIX}subdomain.example.com"
-    redis.set(key, settings.THUMBNAIL_HTTP_ERROR_THRESHOLD_TO_NOTIFY - 1)
-
-    # Call several times
-    for _ in range(3):
-        with pytest.raises(
-            UpstreamThumbnailException, match=r"Failed to render thumbnail."
-        ):
-            photon_get(TEST_IMAGE_URL)
-
-    capture_exception.assert_called_once_with(exc)
-
-    assert redis.get(key) == b"1002"
+    assert_func = (
+        capture_exception.assert_called_once
+        if should_alert
+        else capture_exception.assert_not_called
+    )
+    assert_func()
+    assert redis.get(key) == str(count_start + 1).encode()
 
 
+@alert_count_params
 @pytest.mark.parametrize(
-    "exception, expected_message",
+    "status_code, text",
     [
-        (  # Includes HTTP response errors
-            requests.RequestException(),
-            r"Failed to render thumbnail.",
-        ),
-        (ValueError(), r"due to unidentified exception."),
+        (400, "Bad Request"),
+        (401, "Unauthorized"),
+        (403, "Forbidden"),
+        (500, "Internal Server Error"),
     ],
 )
-def test_get_raise_exception_msg(
-    capture_exception, setup_requests_get_exception, exception, expected_message
+def test_get_http_exception_handles_error(
+    status_code,
+    text,
+    count_start,
+    should_alert,
+    capture_exception,
+    setup_requests_get_exception,
+    redis,
 ):
-    setup_requests_get_exception(exception)
+    mock_response = MagicMock(spec=requests.Response)
+    mock_response.status_code = status_code
+    mock_response.text = text
+    exc = requests.HTTPError(response=mock_response)
+    setup_requests_get_exception(exc)
```

Comment:
Nit: these tests don't need to use `setup_requests_get_exception` and can use `pook` without issue. If they used pook, they would be more explicit/easier to understand and precisely match the actual behaviour of requests for these status codes (without encoding/relying on specific behaviour).

IIRC the only reason the tests above this one use `setup_requests_get_exception` is because the timeout testing in `pook` doesn't work exactly as I expected it to... or I ran into an issue there... I can't remember.

However, there are a number of tests that should be converted to use `pook` rather than mocking requests directly and we should have an issue (similar to https://github.com/WordPress/openverse/issues/1938) for refactoring these. Long-term it produces a more stable test suite, I think.